app/src/org/commcare/android/util/TemplatePrinterUtils.java

```diff
@@ -0,0 +1,184 @@
+package org.commcare.android.util;
+
+import android.text.TextUtils;
+
+import org.commcare.android.crypt.CryptUtil;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+
+/**
+ * Various utilities used by TemplatePrinterTask and TemplatePrinterActivity
+ * 
+ * @author Richard Lu
+ * @author amstone
+ */
+public abstract class TemplatePrinterUtils {
+
+    private static final String FORMAT_REGEX_WITH_DELIMITER = "((?<=%2$s)|(?=%1$s))";
+    private static final SecretKey KEY = CryptUtil.generateSemiRandomKey();
+
+    /**
+     * Gets the file extension from the given file path.
+     *
+     * @param path File path
+     * @return File extension
+     */
+    public static String getExtension(String path) {
```

Comment:
Please avoid obvious javadoc comments. If you rename this method to getFileExtension and the param to filepath, the comments aren't needed.